src/Decoder.elm

```diff
@@ -0,0 +1,38 @@
+module Decoder exposing (..)
+
+import Data.Audience exposing (Audience, AudienceType(..))
+import Data.AudienceFolder exposing (AudienceFolder)
+import Json.Decode exposing (Decoder, andThen, fail, field, int, map3, map4, maybe, string, succeed)
+
+
+audienceFolderDecoder : Decoder AudienceFolder
+audienceFolderDecoder =
+    map3 AudienceFolder
+        (field "id" int)
+        (field "name" string)
+        (maybe (field "parent" int))
```

Comment:
This decoders are perfectly fine, but I'll give you a couple of options to prevent you from relying on `mapN` functions 😉 :

- Use `Decode.andMap` from `json-extra` package, this is the most common way you'll see in all GWI repos.
```elm
audienceFolderDecoder : Decoder AudienceFolder
audienceFolderDecoder =
    Decode.succeed AudienceFolder
        |> Decode.andMap (field "id" int)
        |> Decode.andMap (field "name" string)
        |> Decode.andMap (maybe <| field "parent" int)
```

- Use Richard Feldman's [package](https://package.elm-lang.org/packages/NoRedInk/elm-json-decode-pipeline/latest) for pipeline operator (my personal favourite 🤩)
```elm
audienceFolderDecoder : Decoder AudienceFolder
audienceFolderDecoder =
    Decode.succeed AudienceFolder
        |> required (field "id" int)
        |> required (field "name" string)
        |> required (maybe <| field "parent" int)
```

Reply:
Whoa, it looks much nicer. Didn't know about it. Thanks!